tests/cecil-tests/AttributeTest.cs

```diff
@@ -0,0 +1,198 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Diagnostics;
+
+using NUnit.Framework;
+
+using Mono.Cecil;
+
+#nullable enable
+
+namespace Cecil.Tests {
+
+	// [TestFixture]
+	public class AttributeTest {
+		// https://github.com/xamarin/xamarin-macios/issues/10170
+		// Every binding class that has net6 availability attributes on a method/property
+		// must have one matching every platform listed on the availabilities of the class
+		//
+		// Example:
+		// [SupportedOSPlatform("ios1.0")]
+		// [SupportedOSPlatform("maccatalyst1.0")]
+		// class TestType
+		// {
+		//     public static void Original () { }
+		//
+		//     [SupportedOSPlatform(""ios2.0"")]
+		//     public static void Extension () { }
+		// }
+		//
+		// In this example, Extension is _not_ considered part of maccatalyst1.0
+		// Because having _any_ SupportedOSPlatform implies only the set explicitly set on that member
+		//
+		// This test should find Extension, note that it has an ios attribute,
+		// and insist that some maccatalyst must also be set.
+		// [TestCaseSource (typeof (Helper), "NetPlatformAssemblies")]
+		public void ChildElementsListAvailabilityForAllPlatformsOnParent (string assemblyPath)
+		{
+			var assembly = Helper.GetAssembly (assemblyPath);
+			if (assembly == null) {
+				Assert.Ignore ("{assemblyPath} could not be found (might be disabled in build)");
+				return;
+			}
+
+#if DEBUG
+			Console.WriteLine(assemblyPath);
+#endif
+
+			// Make a list of Availability on parent.
+			// Each child must have an attribute for each availability (Either NotSupported or Supported)
+			HashSet<string> found = new HashSet<string> ();
+			foreach (var prop in Helper.FilterProperties(assembly, a => HasAnyAvailabilityAttribute(a, includeUnsupported: true))) {
+				Process (prop, prop.FullName, prop.DeclaringType, found);
+			}
+			foreach (var meth in Helper.FilterMethods(assembly, a => HasAnyAvailabilityAttribute(a, includeUnsupported: true))) {
+				Process (meth, meth.FullName, meth.DeclaringType, found);
+			}
+#if DEBUG				
+			Assert.That (found.Count, Is.Zero);
+#else
+			Assert.That (found.Count, Is.Zero, string.Join (", ", found));
+#endif
+		}
+
+		void Process (object item, string fullName, TypeDefinition parent, HashSet<string> found)
+		{
+			// XXX - For now skip generated code until associated generator.cs changes are in
+			if (HasCodegenAttribute(item)) {
+				return;
+			}
+// #if DEBUG
+// 			const string Filter = "AppKit";
+// 			if (!fullName.Contains (" " + Filter)) {
+// 				return;
+// 			}
+// #endif
```

Comment:
remote it? or uncomment?

Reply:
I think this is fine to keep until we are done with the test in general

Reply:
I'm planning on landing these just to uncomment out when debugging the mel and generator changes.